state/oci/objectstorage/objectstorage.go

```diff
@@ -147,38 +175,94 @@ func NewOCIObjectStorageStore(logger logger.Logger) *StateStore {
 
 /************** private helper functions. */
 
+func getValue(metadata map[string]string, key string, valueRequired bool) (value string, err error) {
+	if val, ok := metadata[key]; ok && val != "" {
+		return val, nil
+	}
+	if !valueRequired {
+		return "", nil
+	}
+	return "", fmt.Errorf("missing or empty %s field from metadata", key)
+}
+
+func getOptionalBooleanValue(metadata map[string]string, key string) (value bool, err error) {
+	stringValue, _ := getValue(metadata, key, false)
+	if stringValue == "" {
+		stringValue = "false"
+	}
+	value, err = strconv.ParseBool(stringValue)
+	if err != nil {
+		return false, fmt.Errorf("incorrect value %s for %s, should be 'true' or 'false'", stringValue, key)
+	}
+	return value, nil
+}
+
+func getConfigFilePath(meta map[string]string) (value string, err error) {
+	value, _ = getValue(meta, configFilePathKey, false)
+	if strings.HasPrefix(value, "~/") {
+		usr, _ := user.Current()
+		dir := usr.HomeDir
+		value = filepath.Join(dir, value[2:])
+	}
+	if value != "" {
```

Comment:
I prefer to use the absolute path here in case link issues.

Reply:
Hi @daixiang0 , I am afraid I do not understand your comment. Do you mean to say that ~/ should not be supported at all? For clarity's sake, only accept absolute path values?

Reply:
I mean use an absolute clean path, not include relative path for clear.

Reply:
I would like to understand you exactly right. Do you suggest to not allow ~/ as a shorthand for /home/<user>?

Reply:
You got it.

Reply:
Ping @lucasjellema

Reply:
I will fix this in the next commit; note: I need to update the documentation to reflect this change.